From 78c04c67ed40828d0c5a074316190308652b9f23 Mon Sep 17 00:00:00 2001
From: Ryan Blue <blue@apache.org>
Date: Wed, 12 Aug 2015 16:27:44 -0700
Subject: [PATCH 111/176] CLOUDERA-BUILD. Revert PARQUET-162 exception when no
 filter matches.

PARQUET-162 was not backported, but the Thrift converter has changed
substantially and pulled in code that throws an exception when a column
filter doesn't select any columns. This removes the check because it is
a breaking behavior change.
---
 .../java/parquet/thrift/ThriftSchemaConverter.java |    3 ++-
 .../parquet/thrift/TestThriftSchemaConverter.java  |    2 ++
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/parquet-thrift/src/main/java/parquet/thrift/ThriftSchemaConverter.java b/parquet-thrift/src/main/java/parquet/thrift/ThriftSchemaConverter.java
index 3dc4411..9cd1dcd 100644
--- a/parquet-thrift/src/main/java/parquet/thrift/ThriftSchemaConverter.java
+++ b/parquet-thrift/src/main/java/parquet/thrift/ThriftSchemaConverter.java
@@ -63,7 +63,8 @@ public class ThriftSchemaConverter {
     ThriftSchemaConvertVisitor visitor = new ThriftSchemaConvertVisitor(fieldProjectionFilter);
     thriftClass.accept(visitor);
     MessageType convertedMessageType = visitor.getConvertedMessageType();
-    fieldProjectionFilter.assertNoUnmatchedPatterns();
+    // This introduces a behavior change that is not backported, PARQUET-162
+    // fieldProjectionFilter.assertNoUnmatchedPatterns();
     return convertedMessageType;
   }
 
diff --git a/parquet-thrift/src/test/java/parquet/thrift/TestThriftSchemaConverter.java b/parquet-thrift/src/test/java/parquet/thrift/TestThriftSchemaConverter.java
index 59d6504..02190a2 100644
--- a/parquet-thrift/src/test/java/parquet/thrift/TestThriftSchemaConverter.java
+++ b/parquet-thrift/src/test/java/parquet/thrift/TestThriftSchemaConverter.java
@@ -23,6 +23,7 @@ import static org.junit.Assert.fail;
 import static parquet.schema.MessageTypeParser.parseMessageType;
 
 import org.apache.thrift.TBase;
+import org.junit.Ignore;
 import org.junit.Test;
 
 import parquet.schema.MessageType;
@@ -204,6 +205,7 @@ public class TestThriftSchemaConverter {
   }
 
   @Test
+  @Ignore("This tests a behavior change that is not backported, PARQUET-162")
   public void testThrowWhenProjectionFilterMatchesNothing() {
     shouldThrowWhenProjectionFilterMatchesNothing("non_existing", "non_existing", TestStructInMap.class);
     shouldThrowWhenProjectionFilterMatchesNothing("name;non_existing", "non_existing", TestStructInMap.class);
-- 
1.7.9.5

